apollo/handlers/user.py

```diff
@@ -43,24 +43,25 @@ def post_user(user_data: CreateUserSchema,
 def patch_user(user_data: UpdateUserSchema, request: Request,
                session: Session = Depends(get_session)):
     user = request.current_user
-    data = user_data.dict()
-
-    if not compare_plaintext_to_hash(user_data.old_password,
-                                     user.password_hash,
-                                     user.password_salt):
-        return JSONResponse(
-            status_code=400,
-            content={
-                'old_password': {
-                    'msg': 'Invalid password',
-                    'type': 'value_error'
+    data = user_data.dict(exclude_unset=True)
+
+    if data.get('password'):
```

Comment:
I'd move what's under this `if` statement to a different function. It'll increase readability.